paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -1274,4 +1308,101 @@ static ConflictCheck noConflictCheck() {
     public static ConflictCheck mustConflictCheck() {
         return latestSnapshot -> true;
     }
+
+    boolean doCommit(
+            long newSnapshotId,
+            Path newSnapshotPath,
+            Snapshot newSnapshot,
+            boolean useSerializableIsolation,
+            boolean useLock) {
+        try {
+            boolean committed = tryAtomicCommitNewSnapshot(newSnapshotPath, newSnapshot);
+            if (committed) {
+                if (useSerializableIsolation && !useLock) {
+                    // If the useSerializableIsolation is turned on,And we not use lock service,
+                    // we will remove all HINTs,as it may provide incorrect information.
+                    snapshotManager.removeSnapshotEarliestHint();
+                    snapshotManager.removeSnapshotLatestHint();
+                    snapshotManager.removeChangeLogEarliestHint();
+                    snapshotManager.removeChangeLogLatestHint();
+                } else {
+                    snapshotManager.commitLatestHint(newSnapshotId);
+                }
+            }
+            return committed;
+        } catch (Exception e) {
+            throw new CommitStateUnknownException(e.getMessage(), e);
```

Comment:
If any exception occurs during the commit process, we should throw a CommitStateUnknownException and not clean up any data.